bindings/matrix-sdk-ffi/src/room.rs

```diff
@@ -522,3 +561,111 @@ impl TryFrom<ImageInfo> for RumaAvatarImageInfo {
         }))
     }
 }
+
+#[derive(uniffi::Record)]
+/// The power level settings required for various operations within a room.
+/// When updating these settings, any levels that are `None` will remain
+/// unchanged.
+pub struct RoomPowerLevelSettings {
```

Comment:
I'm split on whether these should only be FFI, or whether it makes sense to add them to the SDK too.

Is it possible to create a struct in the SDK that is exported over uniffi yet? It didn't seem worth the effort to map this thing in both directions twice if that wasn't the case.

Reply:
It is possible, but seems to require build system changes. See https://github.com/matrix-org/matrix-rust-sdk/pull/2240. If you could look into that, it would be amazing! (I'm still available for chat on Matrix, just not in any of the internal rooms)